src/UniversalSort.java

```diff
@@ -0,0 +1,41 @@
+package components.UniversalSort;
+
+/**
+ * {@code universalSortKernel} enhanced with secondary methods.
+ */
+public interface UniversalSort extends UniversalSortKernel {
```

Comment:
Contracts look great, but I am a little skeptical of the design. Do these layer correctly?